pdata/internal/generated_immutable_slice.go

```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+
+// Code generated by "model/internal/cmd/pdatagen/main.go". DO NOT EDIT.
+// To regenerate this file run "go run model/internal/cmd/pdatagen/main.go".
+
+package internal
+
+// ImmutableByteSlice represents a []byte slice that cannot be mutated.
+type ImmutableByteSlice struct {
+	value []byte
+}
+
+// NewImmutableByteSlice creates a new ImmutableByteSlice by moving an underlying array of the provided []byte slice.
+// Note that the argument value will be cleared after the call,
+// use NewImmutableByteSliceFromCopy if it needs to be preserved.
+func NewImmutableByteSlice(val *[]byte) ImmutableByteSlice {
+	is := ImmutableByteSlice{}
+	if val != nil {
+		is.value = *val
+		*val = nil
+	}
+	return is
+}
+
+// NewImmutableByteSliceFromCopy creates a new ImmutableByteSlice by copying the provided []byte slice.
+// Use NewImmutableByteSlice if the provided slice is not needed anymore.
+func NewImmutableByteSliceFromCopy(val []byte) ImmutableByteSlice {
+	is := ImmutableByteSlice{}
+	if len(val) != 0 {
+		is.value = make([]byte, len(val))
+		copy(is.value, val)
+	}
+	return is
+}
+
+// Value returns a copy of the []byte slice value.
+func (is ImmutableByteSlice) Value() []byte {
+	if len(is.value) == 0 {
+		return nil
+	}
+	val := make([]byte, len(is.value))
+	copy(val, is.value)
+	return val
+}
+
+// ImmutableDoubleSlice represents a []float64 slice that cannot be mutated.
+type ImmutableDoubleSlice struct {
+	value []float64
+}
+
+// NewImmutableDoubleSlice creates a new ImmutableDoubleSlice by moving an underlying array of the provided []float64 slice.
+// Note that the argument value will be cleared after the call,
+// use NewImmutableDoubleSliceFromCopy if it needs to be preserved.
+func NewImmutableDoubleSlice(val *[]float64) ImmutableDoubleSlice {
+	is := ImmutableDoubleSlice{}
+	if val != nil {
+		is.value = *val
+		*val = nil
+	}
+	return is
+}
+
+// NewImmutableDoubleSliceFromCopy creates a new ImmutableDoubleSlice by copying the provided []float64 slice.
+// Use NewImmutableDoubleSlice if the provided slice is not needed anymore.
+func NewImmutableDoubleSliceFromCopy(val []float64) ImmutableDoubleSlice {
+	is := ImmutableDoubleSlice{}
+	if len(val) != 0 {
+		is.value = make([]float64, len(val))
+		copy(is.value, val)
+	}
+	return is
+}
+
+// Value returns a copy of the []float64 slice value.
+func (is ImmutableDoubleSlice) Value() []float64 {
+	if len(is.value) == 0 {
+		return nil
+	}
+	val := make([]float64, len(is.value))
+	copy(val, is.value)
+	return val
+}
+
+// ImmutableUIntSlice represents a []uint64 slice that cannot be mutated.
+type ImmutableUIntSlice struct {
+	value []uint64
+}
+
+// NewImmutableUIntSlice creates a new ImmutableUIntSlice by moving an underlying array of the provided []uint64 slice.
+// Note that the argument value will be cleared after the call,
+// use NewImmutableUIntSliceFromCopy if it needs to be preserved.
+func NewImmutableUIntSlice(val *[]uint64) ImmutableUIntSlice {
```

Comment:
Uint64 in the name in case we ever need the real uint?

Reply:
Yes, makes sense. But we still keep `NewImmutableDoubleSlice(val *[]float64)`, right?

Reply:
For the "primitives" slices I would go with the go terminology.